androidHyperskillApp/src/main/java/org/hyperskill/app/android/core/view/ui/widget/compose/LinearProgressIndicator.kt

```diff
@@ -0,0 +1,92 @@
+package org.hyperskill.app.android.core.view.ui.widget.compose
+
+import androidx.compose.foundation.Canvas
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.progressSemantics
+import androidx.compose.material.ProgressIndicatorDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.Immutable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.graphics.Brush
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.drawscope.DrawScope
+import androidx.compose.ui.unit.LayoutDirection
+import androidx.compose.ui.unit.dp
+
+/**
+ * Determinate <a href="https://material.io/components/progress-indicators#linear-progress-indicators" class="external" target="_blank">Material Design linear progress indicator</a>.
+ *
+ * **This is a copy of [androidx.compose.material.LinearProgressIndicator] with the ability to set gradient color of the progress.**
+ *
+ * Progress indicators express an unspecified wait time or display the length of a process.
+ *
+ * By default, there is no animation between [progress] values. You can use
+ * [ProgressIndicatorDefaults.ProgressAnimationSpec] as the default recommended
+ * [AnimationSpec] when animating progress.
+ *
+ * @param progress The progress of this progress indicator, where 0.0 represents no progress and 1.0
+ * represents full progress. Values outside of this range are coerced into the range.
+ * @param brush The gradient color of the progress indicator.
+ * @param backgroundColor The color of the background behind the indicator, visible when the
+ * progress has not reached that area of the overall indicator yet.
+ */
+@Composable
+fun LinearProgressIndicator(
+    /*@FloatRange(from = 0.0, to = 1.0)*/
+    progress: Float,
+    modifier: Modifier = Modifier,
+    brush: Brush,
+    backgroundColor: Color
+) {
+    Canvas(
+        modifier
+            .progressSemantics(progress)
+            .size(LinearIndicatorWidth, LinearIndicatorHeight)
```

Comment:
This is a copy of the MaterialDiasing lib component to use a brush instead of regular color. It has the max width and predefined height according to the material spec.